lib/game.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'set'
+
+class Game
+  @wins = [[1, 2, 3], [1, 4, 7], [1, 5, 7], [2, 5, 8], [3, 6, 9], [3, 5, 7], [4, 5, 6], [7, 8, 9]]
+  @options = []
+
+  def initialize(player1, player2, board)
+    @player1 = player1
+    @player2 = player2
+    @board = board
+    @wins = [[1, 2, 3], [1, 4, 7], [1, 5, 7], [2, 5, 8], [3, 6, 9], [3, 5, 7], [4, 5, 6], [7, 8, 9]]
+    @options = []
+  end
+
+  def game_start
+    a = 0
+
+    until board_full?
+      @board.display
+      option = 0
+
+      current_player = a.even? ? @player1 : @player2
+
+      loop do
+        puts " #{current_player}, select an option from 1..9"
+        option = gets.chomp.to_i
+
+        break if valid_move?(option)
+      end
+
+      char = (a.even? ? 'X' : 'O').to_s
+      @board.grid[option - 1] = char if @board.grid.include?(option)
+
+      @wins = @wins.each do |moves|
+        moves[moves.find_index(option)] = char if moves.include?(option)
```

Comment:
This is clearly not working, you see when the first player has a winning position, i.e. `[1, 5, 9]`, as such:

```
X|O|3
----- ----- -----
4|X|6
----- ----- -----
O|8|9
```

And player 1 makes the move to `9`, your `@wins` instance variable has the following moves stored:
```
(byebug) @wins
[["X", "O", 3], ["X", 4, "O"], ["X", "X", "O"], ["O", "X", 8], [3, 6, 9], [3, "X", "O"], [4, "X", 6], ["O", 8, 9]]
```

I don't see a triple `X` in there and that is why it isn't recognizing the winner.

You can isolate this on its own and test it by setting `@wins` to whatever you want and see if it determines a winner. Once you have that, you know for sure that this works **if it gets a correct `@wins`** array of arrays; however, from the looks of it, it isn't. Nevertheless, if you check your `@board.grid` the moves are correctly stored there:

```
(byebug) @board.grid
["X", "O", 3, 4, "X", 6, "O", 8, "X"]
```

You can also try to debug it to see what is going on by using `byebug`, all you have to do is `require 'byebug'` at the top of your file and then insert the statement `byebug` where you want to set the breakpoint, then you can use the `s` and `n` commands to step into your code and check the state of your variables at all times.

I have something to ask you though, why are you storing the moves in both `@board.grid` and `@wins`? Surely there must be a better way to go about it. Nevertheless I realize the effort spent, so you just might want to try and figure out why it isn't working to get a working solution out.